### ISSUE TITLE
[M-3] Partial junior exit builds phantom residual that prematurely reopens DepositJunior gate during active loss

### ISSUE THREAD

**Issue by v1ktorrr0x** [dcccrypto/percolator-stake#222]
# [M-3] Partial junior exit builds phantom residual that prematurely reopens `DepositJunior` gate during active loss

**Severity:** Medium
**File:** `src/processor.rs` (~L1187–1208, L2448)

---

## Description

During an active insurance loss (`total_flushed > total_returned`), the junior withdrawal amount is priced against `effective_junior_balance()` — the loss-adjusted value, which is smaller than the raw proportional share. However, the raw `junior_balance()` is reduced by that smaller value:

```rust
pool.set_junior_balance(
    pool.junior_balance()
        .checked_sub(withdrawal_amount)   // withdrawal_amount < raw proportional share
        .ok_or(StakeError::Overflow)?
);
```

Because `withdrawal_amount < proportional share of raw junior_balance`, raw balance decreases by less than the effective share consumed. This phantom residual compounds across multiple partial exits.

When the final junior exits and triggers the `new_junior_lp == 0` branch, `forfeited = junior_balance() - effective_junior_balance()` captures the entire accumulated residual and adds it to `total_returned`. If the residual is large enough, `total_returned` can equal or exceed `total_flushed`, satisfying:

```
total_flushed <= total_returned
```

…before any physical insurance return has occurred. The `DepositJunior` gate reads only raw `total_returned`:

```rust
// processor.rs L2448
if pool.total_flushed > pool.total_returned {
    return Err(StakeError::InsuranceLossOutstanding.into());
}
```

This gate passes, allowing new junior depositors to enter during an unresolved loss.

## Impact

New junior depositors can inherit a pre-existing insurance loss they were never exposed to. The direction of value transfer is from new junior depositors to existing senior holders who are protected from the loss, violating the guarantee the gate is designed to enforce.

## Recommended Fix

The gate must reflect only physically returned funds, not bookkeeping settlements. Factor in `realized_junior_loss`:

```rust
let physical_net_loss = pool.total_flushed
    .saturating_sub(
        pool.total_returned.saturating_sub(pool.realized_junior_loss())
    );
if physical_net_loss > 0 {
    return Err(StakeError::InsuranceLossOutstanding.into());
}
```

Equivalently: `total_flushed > total_returned + realized_junior_loss` as the gate condition.

## Notes

This finding is coupled with **M-1**. Both are fixed by the same `realized_junior_loss` correction term. The root cause is that `realized_junior_loss` is correctly accounted for in LP pricing math (`total_pool_value`) but is absent from the two control-flow checks that gate insurance recovery and new junior admission.

**Comment by dcccrypto** [comment]
Fixed by #225 (merged to main — tranche-accounting M-1/M-2/M-3 + L-1 vault owner check + L-3 named accessor; verified, 138 lib + full integration suite green). Thanks @v1ktorrr0x.
